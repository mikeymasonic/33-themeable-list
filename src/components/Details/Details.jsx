import React from 'react';
import PropTypes from 'prop-types';

const Details = ({ name, image, species, status, origin, location }) => (
  <>
    <h3>{name}</h3>
    {image && <img src={image} />}
    {species && <h4>Species: {species}</h4>}
    {status && <h4>Status: {status}</h4>}
    {origin && <h4>Origin: {origin}</h4>}
    {location && <h4>Location: {location}</h4>}
  </>
);

Details.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  species: PropTypes.string,
  status: PropTypes.string,
  origin: PropTypes.string,
  location: PropTypes.string
};

export default Details;
