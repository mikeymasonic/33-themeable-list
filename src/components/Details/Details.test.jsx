import React from 'react';
import { shallow } from 'enzyme';
import Details from './Details';

describe('Details component', () => {
  it('renders Details', () => {
    const wrapper = shallow(<Details 
      name="Rick Sanchez" 
      species="Human"
      status="Alive"
      image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      origin="Earth (C-137)"
      location="Earth (Replacement Dimension)"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
