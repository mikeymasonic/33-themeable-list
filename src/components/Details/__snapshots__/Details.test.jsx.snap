// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `
<Fragment>
  <h3>
    Rick Sanchez
  </h3>
  <img
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <h4>
    Species: 
    Human
  </h4>
  <h4>
    Status: 
    Alive
  </h4>
</Fragment>
`;
